AllisonsCatPalaceTDDTests/LocationControllerTests.swift

```diff
@@ -0,0 +1,332 @@
+
+        XCTAssertFalse(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(), replacementString: "a"),
+                       "Zip code field should not accept alphabetic input")
+
+        XCTAssertFalse(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(), replacementString: "!"),
+                       "Zip code field should not accept special character input")
+
+        XCTAssertFalse(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(), replacementString: "@"),
+                       "Zip code field should not accept special character input")
+
+        XCTAssertFalse(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(), replacementString: "."),
+                       "Zip code field should not accept special character input")
+
+        XCTAssertFalse(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(), replacementString: " "),
+                       "Zip code field should not accept white space input")
+
+        XCTAssertFalse(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(), replacementString: "\n"),
+                       "Zip code field should not accept newline input")
+    }
+
+    func testZipCodeFieldCharacterLimit() {
+        textField.text = ""
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(), replacementString: "8"))
+
+        textField.text = "8"
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 1, length: 0), replacementString: "0"))
+
+        textField.text = "80"
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 2, length: 0), replacementString: "2"))
+
+        textField.text = "802"
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 3, length: 0), replacementString: "2"))
+
+        textField.text = "8022"
+        XCTAssertTrue(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 4, length: 0), replacementString: "0"))
+
+        textField.text = "80220"
+        XCTAssertFalse(delegate.textField!(textField, shouldChangeCharactersIn: NSRange(location: 5, length: 0), replacementString: "8"))
+    }
+
+    func testZipCodeFieldBeginsEditing() {
+        // resets borderColor
+        textField.layer.borderColor = UIColor.green.cgColor
+        XCTAssertEqual(textField.layer.borderColor, UIColor.green.cgColor,
+                       "Zip code field's border color should be green")
+
+        delegate.textFieldDidBeginEditing?(textField)
+        XCTAssertEqual(textField.layer.borderColor, UIColor.lightGray.cgColor,
+                       "Zip code field's border color should be reset to light gray on beginning text editing")
+
+        // stops spinner
+        controller.activityIndicator.startAnimating()
+        XCTAssertTrue(controller.activityIndicator.isAnimating,
```

Comment:
take this out. Will know that it's animating if I told UIKit to start animating.